blog/migrations/0019_migrate_image_data_to_custom_image_model.py

```diff
@@ -0,0 +1,50 @@
+from django.conf import settings
+from django.db import migrations
+
+
+def move_existing_image_data_to_custom_model(apps, _schema_editor):
+    wagtail_image_model = apps.get_model("wagtailimages", "image")
+    custom_image_model = apps.get_model("blog", "customimage")
+    custom_image_model.objects.bulk_create([
+        custom_image_model(
+            title=image.title,
+            file=image.file,
+            width=image.width,
+            height=image.height,
+            created_at=image.created_at,
+            uploaded_by_user=image.uploaded_by_user,
+            tags=image.tags,
+            focal_point_x=image.focal_point_x,
+            focal_point_y=image.focal_point_y,
+            focal_point_width=image.focal_point_width,
+            focal_point_height=image.focal_point_height,
+            file_size=image.file_size,
+            file_hash=image.file_hash,
+        ) for image in wagtail_image_model.objects.all().order_by("pk")
+    ])
+
+
+def assign_new_images_to_articles(apps, _schema_editor):
+    custom_image_model = apps.get_model("blog", "customimage")
+    blog_article_page_model = apps.get_model("blog", "blogarticlepage")
+    for blog_article_page in blog_article_page_model.objects.all():
+        blog_article_page.new_cover_photo = custom_image_model.objects.get(pk=blog_article_page.cover_photo.pk)
+        blog_article_page.new_article_photo = custom_image_model.objects.get(pk=blog_article_page.article_photo.pk)
+        for stream_child in blog_article_page.body:
+            if stream_child.block.name == "image":
+                original_image = stream_child.value
+                stream_child.value = custom_image_model.objects.get(pk=original_image.pk)
```

Comment:
I've tested it by switching images at the last line. It did work after the migration was applied.

Reply:
Although replacing images in the body seems to already work perfectly without this code. Should I leave this explicit operation here or should we fall back on Wagtail's default behavior, which does what we want?

Reply:
Yup, you can leave it

Reply:
Ok. I'll test it again before merging 👌